test/files/pos/t8855.scala

```diff
@@ -0,0 +1,13 @@
+object Test {
+  final val clasz = classOf[String]
+  final val str = "Str"
+
+  import java.lang.annotation.RetentionPolicy.CLASS.{ordinal => ord} // error: stable identifier required, but CLASS found
+  println(ord)
+
+  import clasz.{getName => claszName} // error: stable identifier required, but classOf[java.lang.String] found
+  println(claszName)
+
+  import str.{length => strLen} // error: stable identifier required, but "Str" found
+  println(strLen)
+}
```

Comment:
You promised me a rose garden.
```
scala> final val x = 42
val x: Int(42) = 42

scala> import x._
java.lang.NullPointerException
	at scala.tools.nsc.typechecker.Contexts.registerImport(Contexts.scala:92)
	at scala.tools.nsc.typechecker.Contexts.registerImport$(Contexts.scala:90)
	at scala.tools.nsc.Global$$anon$5.registerImport(Global.scala:483)
	at scala.tools.nsc.typechecker.Namers$Namer.scala$tools$nsc$typechecker$Namers$Namer$$importSig(Namers.scala:1812)
	at scala.tools.nsc.typechecker.Namers$Namer$ImportTypeCompleter.completeImpl(Namers.scala:903)
```

Reply:
I stashed that fix at https://github.com/scala/scala/pull/8627 for when this commit arrives in 2.13.